SerialPrograms/Source/PokemonLA/Programs/PokemonLA_RegionNavigation.cpp

```diff
@@ -380,6 +380,94 @@ void goto_camp_from_overworld(
     context.wait_for(std::chrono::milliseconds((uint64_t)(GameSettings::instance().POST_WARP_DELAY * 1000)));
 }
 
+void goto_any_camp_from_overworld(ProgramEnvironment& env, ConsoleHandle& console, BotBaseContext& context,
+                                  const TravelLocation& location){
+
+    auto start = current_time();
+    std::chrono::seconds grace_period(0);
+    while (true){
+        {
+            EscapeFromAttack session(
+                env, console, context,
+                grace_period, std::chrono::seconds(10)
+            );
+            session.run_session();
+        }
+
+        if (current_time() - start > std::chrono::seconds(60)){
+            dump_image(console.logger(), env.program_info(), "EscapeFromAttack", console.video().snapshot());
+            throw OperationFailedException(console, "Unable to escape from being attacked.");
+        }
+
+        //  Open the map.
+        pbf_press_button(context, BUTTON_MINUS, 20, 30);
+        {
+            MapDetector detector;
+            int ret = wait_until(
+                console, context,
+                std::chrono::seconds(5),
+                {{detector}}
+            );
+            if (ret < 0){
+                dump_image(env.logger(), env.program_info(), "MapNotFound", console.video().snapshot());
+                throw OperationFailedException(console, "Map not detected after 5 seconds.");
+            }
+            console.log("Found map!");
+            context.wait_for(std::chrono::milliseconds(500));
+        }
+
+        //  Warp to sub-camp.
+        pbf_press_button(context, BUTTON_X, 20, 30);
+        {
+            ButtonDetector detector(
+                console, console,
+                ButtonType::ButtonA,
+                {0.55, 0.40, 0.20, 0.40},
+                std::chrono::milliseconds(200), true
+            );
+            int ret = wait_until(
+                console, context,
+                std::chrono::seconds(2),
+                {{detector}}
+            );
+            if (ret >= 0){
+                console.log("Flying back to camp...");
+                pbf_wait(context, 50);
+                if (location.warp_slot != 0){
+                    for (size_t c = 0; c < location.warp_slot; c++){
+                        pbf_press_dpad(context, DPAD_DOWN, 20, 30);
+                    }
+                }
+                for (size_t c = 0; c < location.warp_sub_slot; c++){
+                    pbf_press_dpad(context, DPAD_DOWN, 20, 30);
+                }
+                pbf_mash_button(context, BUTTON_A, 125);
+                break;
+            }
+
+            if (ret < 0){
+                throw OperationFailedException(console, "Unable to fly. Are you under attack?");
```

Comment:
Is this intentional? `EscapeFromAttack` is not reliable in that when it returns you may still be under attack. Thus the loop that you probably based this from is a loop so that it can retry if it still can't fly.

Reply:
yeah, removed, I think was a left over from when I was playing around with it!